python/test/Ice/exceptions/TestAMDI.py

```diff
@@ -0,0 +1,183 @@
+    def throwBasB(self, a: int, b: int, current: Ice.Current):
+        ex = Test.B()
+        ex.aMem = a
+        ex.bMem = b
+        raise ex
+
+    @override
+    def throwCasB(self, a: int, b: int, c: int, current: Ice.Current):
+        ex = Test.C()
+        ex.aMem = a
+        ex.bMem = b
+        ex.cMem = c
+        f = Ice.Future()
+        f.set_exception(ex)
+        return f
+
+    @override
+    def throwCasC(self, a: int, b: int, c: int, current: Ice.Current):
+        ex = Test.C()
+        ex.aMem = a
+        ex.bMem = b
+        ex.cMem = c
+        f = Ice.Future()
+        f.set_exception(ex)
+        return f
+
+    @override
+    def throwModA(self, a: int, a2: int, current: Ice.Current):
+        ex = Mod.A()
+        ex.aMem = a
+        ex.a2Mem = a2
+        raise ex
+
+    @override
+    def throwUndeclaredA(self, a: int, current: Ice.Current):
+        ex = Test.A()
+        ex.aMem = a
+        f = Ice.Future()
+        f.set_exception(ex)
+        return f
+
+    @override
+    def throwUndeclaredB(self, a: int, b: int, current: Ice.Current):
+        ex = Test.B()
+        ex.aMem = a
+        ex.bMem = b
+        raise ex
+
+    @override
+    def throwUndeclaredC(self, a: int, b: int, c: int, current: Ice.Current):
+        ex = Test.C()
+        ex.aMem = a
+        ex.bMem = b
+        ex.cMem = c
+        f = Ice.Future()
+        f.set_exception(ex)
+        return f
+
+    @override
+    def throwLocalException(self, current: Ice.Current):
+        f = Ice.Future()
+        f.set_exception(Ice.TimeoutException())
+        return f
+
+    @override
+    def throwNonIceException(self, current: Ice.Current):
+        f = Ice.Future()
+        f.set_exception(RuntimeError("12345"))
+        return f
+
+    @override
+    def throwAssertException(self, current: Ice.Current):
+        raise RuntimeError("operation `throwAssertException' not supported")
+
+    @override
+    def throwMemoryLimitException(self, seq: bytes, current: Ice.Current):
+        return Ice.Future.completed(bytearray(20 * 1024))
+
+    @override
+    def throwLocalExceptionIdempotent(self, current: Ice.Current):
+        f = Ice.Future()
+        f.set_exception(Ice.TimeoutException())
+        return f
+
+    @override
+    def throwDispatchException(self, replyStatus: int, current: Ice.Current):
+        f = Ice.Future()
+        f.set_exception(Ice.DispatchException(replyStatus))
+        return f
+
+    @override
+    def throwAfterResponse(self, current: Ice.Current):
+        # Cannot be implemented with Futures
+        return None
+
+    @override
+    def throwAfterException(self, current: Ice.Current):
+        # Cannot be implemented with Futures
+        f = Ice.Future()
+        f.set_exception(Test.A())
+        return f
+
+    @override
+    def throwMarshalException(self, current: Ice.Current) -> Awaitable[tuple[int, int]]:
+        # We return the wrong type on purpose to test that we throw MarshalException.
```

Comment:
Don't we need to disable type checks here to be able to return the wrong type?

Reply:
Pyright doesn't care. I think because of the signature of the Ice.Futre func. Something we should look into improving.